### PR TITLE
Fixes bileworms being unable to act

### LINKED ISSUE #69608: Bileworms broken
Reporting client version: 514.1584

## Round ID:
[189607](https://scrubby.melonmesa.com/round/189607)

## Testmerges:
- [Converts SSblackbox feedback to an associative list](https://www.github.com/tgstation/tgstation/pull/69399)

## Reproduction:
https://www.youtube.com/watch?v=dIUzu9KIC6c
if you dont wanna watch video,  they dont move, spit or anything, they TRY to spit, but dont

### PR DESCRIPTION
**Pull request by GoblinBackwards**
## About The Pull Request
Fixes an issue where bileworms couldn't do anything as their code assumed their target would be a direct reference to the mob, but #69253 changed basic mob AI to use weakrefs instead and didn't update the bileworm code to handle this.

Fixes #69608

## Why It's Good For The Game
bugfix

## Changelog

:cl:
fix: Fixed bileworms being unable to act.
/:cl:

### CODE REVIEW

File: code/datums/ai/basic_mobs/basic_ai_behaviors/try_mob_ability.dm [hunk 1]
```diff
@@ -12,7 +13,8 @@
 
 /datum/ai_behavior/try_mob_ability/finish_action(datum/ai_controller/controller, succeeded, ability_key, target_key)
 	. = ..()
-	var/mob/living/target = controller.blackboard[target_key]
+	var/datum/weakref/weak_target = controller.blackboard[target_key]
+	var/mob/living/target = weak_target.resolve()
 	if(!target || QDELETED(target) || target.stat >= UNCONSCIOUS)
```
Comment (ShizCalev) [inline code/datums/ai/basic_mobs/basic_ai_behaviors/try_mob_ability.dm:18]:
```suggestion
	if(QDELETED(target) || target.stat >= UNCONSCIOUS)
```

File: code/modules/mob/living/basic/lavaland/bileworm/bileworm_ai.dm [hunk 2]
```diff
@@ -13,7 +13,8 @@
 
 /datum/ai_planning_subtree/bileworm_attack/SelectBehaviors(datum/ai_controller/controller, delta_time)
 
-	var/atom/target = controller.blackboard[BB_BASIC_MOB_CURRENT_TARGET]
+	var/datum/weakref/weak_target = controller.blackboard[BB_BASIC_MOB_CURRENT_TARGET]
+	var/mob/living/target = weak_target.resolve()
 	if(!target || QDELETED(target))
```
Comment (ShizCalev) [inline code/modules/mob/living/basic/lavaland/bileworm/bileworm_ai.dm:18]:
```suggestion
	if(QDELETED(target))
```

File: code/modules/mob/living/basic/lavaland/bileworm/bileworm_ai.dm [hunk 3]
```diff
@@ -31,7 +32,8 @@
 
 /datum/ai_planning_subtree/bileworm_execute/SelectBehaviors(datum/ai_controller/controller, delta_time)
 
-	var/mob/living/target = controller.blackboard[BB_BASIC_MOB_EXECUTION_TARGET]
+	var/datum/weakref/weak_target = controller.blackboard[BB_BASIC_MOB_EXECUTION_TARGET]
+	var/mob/living/target = weak_target.resolve()
 	if(!target || QDELETED(target) || target.stat < UNCONSCIOUS)
```
Comment (ShizCalev) [inline code/modules/mob/living/basic/lavaland/bileworm/bileworm_ai.dm:37]:
```suggestion
	if(QDELETED(target) || target.stat < UNCONSCIOUS)
```

File: code/datums/ai/basic_mobs/basic_ai_behaviors/try_mob_ability.dm [hunk 4]
```diff
@@ -3,7 +3,8 @@
 /datum/ai_behavior/try_mob_ability/perform(delta_time, datum/ai_controller/controller, ability_key, target_key)
 
 	var/datum/action/cooldown/mob_cooldown/ability = controller.blackboard[ability_key]
-	var/mob/living/target = controller.blackboard[target_key]
+	var/datum/weakref/weak_target = controller.blackboard[target_key]
+	var/mob/living/target = weak_target.resolve()
 	if(!ability || !target)
```
Comment (ShizCalev) [inline code/datums/ai/basic_mobs/basic_ai_behaviors/try_mob_ability.dm:8]:
```suggestion
	if(!ability || QDELETED(target))
```

**Review by ShizCalev — APPROVED** [review]

File: code/datums/ai/basic_mobs/basic_ai_behaviors/try_mob_ability.dm [hunk 5]
```diff
@@ -3,17 +3,19 @@
 /datum/ai_behavior/try_mob_ability/perform(delta_time, datum/ai_controller/controller, ability_key, target_key)
 
 	var/datum/action/cooldown/mob_cooldown/ability = controller.blackboard[ability_key]
```
Comment (ShizCalev) [inline code/datums/ai/basic_mobs/basic_ai_behaviors/try_mob_ability.dm:5]: Are these weakrefs too by chance?

Comment (GoblinBackwards) [reply code/datums/ai/basic_mobs/basic_ai_behaviors/try_mob_ability.dm:5]: The abilities aren't weakrefs no

**Pull request merged**